### PR TITLE
Add hash unit tests.

### PR DESCRIPTION
**Pull request by Icantjuddle**
Long time user wanted to start contributing so figured I could start with some low hanging fruit like this to learn the codebase.

Unit test cases are generated from https://github.com/PeterScott/murmur3

### CODE REVIEW

File: src/hash.cc [hunk 1]
```diff
@@ -67,4 +72,16 @@ size_t hash_data(const char* input, size_t len)
     return hash;
 }
 
+UnitTest test_murmer_hash{[] {
```
Comment (mawww) [inline src/hash.cc:75]: murmer ?

File: src/hash.cc [hunk 2]
```diff
@@ -50,10 +53,12 @@ size_t hash_data(const char* input, size_t len)
...
     uint32_t key = 0;
-    switch (len & 3)
+    switch (len & 0b11)
     {
-        case 3: key ^= tail[2] << 16; /* fallthrough */
-        case 2: key ^= tail[1] << 8;  /* fallthrough */
+        case 3: key ^= tail[2] << 16;
+            [[fallthrough]];
+        case 2: key ^= tail[1] << 8;
+            [[fallthrough]];
```
Comment (mawww) [inline src/hash.cc:61]: This alignment is inconsistent with how `case 1` aligns statements, which itself is inconsistent with how the code is usually formatted in Kakoune.

Could you make all cases use consistent indentation (either by moving the fallthrough attirbutes back onto the previous line, keeping all operations clearly aligned, or by introducing a line break after all `case N:` and just indenting by 4 spaces as usual.

**Review by mawww — CHANGES_REQUESTED** [review]
Thanks for that pull request, can you fix the small style issues ?

File: src/hash.cc (hunk 2 above)
Comment (Icantjuddle) [reply src/hash.cc]: Yeah, my b used to there just being a clang-format I can blindly apply :laughing:

**Pull request merged**